### PR TITLE
Update the react native sample to use the latest builder sdk

### PR DESCRIPTION
**Pull request by kylefowler**
## Description

Updates the react native sample app on a few major dependencies
- Updates the builder react-native sample app to the latest version
- Updates expo to the latest
- migrates from react navigation to expo-router
- commits the shell android and ios projects generated by `expo run:ios` and `expo run:android`

TODO:
- After this update the web version of the RN sample will not scroll. Will follow up on that in a future ticket as the mobile sides are working correctly.

### CODE REVIEW

File: examples/react-native/babel.config.js [hunk 1]
```diff
@@ -3,4 +3,4 @@ module.exports = function (api) {
   return {
     presets: ['babel-preset-expo'],
   };
-};
+};
```
Comment (teleaziz) [inline examples/react-native/babel.config.js:6]: there are few linting issues, you probably can fix them all at once using the linter from the root of the repo:
```
yarn lint:fix
```

**Review by teleaziz — COMMENTED** [review]
are those `android` and `ios` foldeers auto generated? if so, maybe we can omit them and simplify this example a little

File: examples/react-native/babel.config.js (hunk 1 above)
Comment (kylefowler) [reply examples/react-native/babel.config.js:6]: should the build fail if there are linting issues?

Comment (kylefowler) [reply examples/react-native/babel.config.js:6]: ah the examples directory isnt part of the root project

Comment (kylefowler) [reply examples/react-native/babel.config.js:6]: got the linter to run off the same config as in the root, it says there are no issues in here

**Review by teleaziz — APPROVED** [review]

**Pull request merged**